src/core/Element.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <QDebug>
+#include <QString>
+
+class Element
+{
+  public:
+    Element() = default;
+    explicit Element(QString descr, int addr);
+    Element(const Element &) = default;
+    Element(Element &&)      = default;
+    virtual ~Element()       = default;
+
+    Element &operator=(const Element &) = default;
+    Element &operator=(Element &&) = default;
+
+    bool operator==(const Element &other) const;
+    bool operator!=(const Element &other) const;
+
+    QString description() const noexcept;
+    virtual int address() const noexcept;
+
+    virtual void setValue(int8_t val)
+    {
+        (void)(val);
+    }
+    virtual void setValue(int16_t val)
+    {
+        (void)(val);
+    }
```

Comment:
This is bug-prone; since the inherited don't need to overload those functions, in the case where you are using _setValue_ with a parameter that is not overloaded you risk to not set the value.
I suggest setting these function to pure virtual and let the other classes reimplement these in the correct way:
```
int16_t m_val
void setValue(int8_t val) {
  m_val = val;
}

int8_t m_val
void setValue(int16_t val) {
  m_val = static_cast<int8_t>(val);
}
```